centos_ci.sh

```diff
@@ -22,79 +22,155 @@ set -e
+  yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+  yum makecache fast
+  yum install -y docker-ce
+  systemctl start docker
+
+  echo 'CICO: Docker installed and started'
+}
+
+function install_and_setup_golang() {
+  # Install Go 1.7
+  curl -LO https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz
+  tar -xf go1.7.linux-amd64.tar.gz
+  mv go /usr/local
+
+  # Setup GOROOT
+  export GOROOT=/usr/local/go
+  # Setup GOPATH
+  mkdir $HOME/gopath $HOME/gopath/src $HOME/gopath/bin $HOME/gopath/pkg
+  export GOPATH=$HOME/gopath
+  export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+}
+
+function setup_repo() {
+  # Setup minishift repo
+  mkdir -p $GOPATH/src/github.com/minishift
+  ln -s `pwd` $GOPATH/src/github.com/minishift/minishift
+}
+
+function setup_glide() {
+  # Setup Glide ("curl https://glide.sh/get | sh" not working due to https://github.com/Masterminds/glide/issues/708)
+  GLIDE_OS_ARCH=`go env GOHOSTOS`-`go env GOHOSTARCH`
+  GLIDE_TAG=v0.12.3
+  GLIDE_LATEST_RELEASE_URL="https://github.com/Masterminds/glide/releases/download/${GLIDE_TAG}/glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz"
+  curl -LO ${GLIDE_LATEST_RELEASE_URL}
+  mkdir /tmp/glide
+  tar --directory=/tmp/glide -xvf glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz
+  export PATH=$PATH:/tmp/glide/${GLIDE_OS_ARCH}
+}
 
-set +x
-# For PR build, GIT_BRANCH is set to branch name other than origin/master
-if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+function prepare() {
+  install_and_setup_golang;
+  setup_repo;
+  setup_glide;
+
+  echo "CICO: Preparation complete"
+}
+
+function run_tests() {
+  cd $GOPATH/src/github.com/minishift/minishift
+  make clean test cross fmtcheck prerelease
+  # Run integration test with 'kvm' driver
+  MINISHIFT_VM_DRIVER=kvm make integration
+
+  echo "CICO: Tests ran successfully"
+}
+
+function artifacts_upload_on_pr_and_master_trigger() {
+  set +x
+  # For PR build, GIT_BRANCH is set to branch name other than origin/master
+  if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/master/$BUILD_NUMBER/
+    cp -r out/* minishift/master/$BUILD_NUMBER/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+  else
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/pr/$ghprbPullId/
+    cp -r out/* minishift/pr/$ghprbPullId/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/pr/$ghprbPullId/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/pr/$ghprbPullId ."
+  fi
+}
+
+function docs_artifacts_upload() {
+  set +x
+
+  version=$(cat docs/build/variables.adoc | cut -d' ' -f2)
```

Comment:
Well, the image build works, but then of course the permissions are screwed up. One way to this this would be to actually create a user on the slave side as well. Same as we do in the Dockerfile. And then we switch to this user. I think that might be a good idea.